### PR TITLE
Solve performance issues when loading folders

### LINKED ISSUE #86: App crashed trying to open HOME folder on my beams account
```python
====== Selected file: mda_0001.mda
Mode is set to Auto-off

Responder: action='clear'

doPlot called: args=('clear', {'Y': [2], 'X': 1})

Responder: action='add'

doPlot called: args=('add', {'Y': []})
Traceback (most recent call last):
  File "/home/beams3/RODOLAKIS/src/mdaviz/mdaviz/mda_folder.py", line 279, in doPlot
    datasets, plot_options = to_datasets(
                             ^^^^^^^^^^^^
  File "/home/beams3/RODOLAKIS/src/mdaviz/mdaviz/select_fields_table_view.py", line 201, in to_datasets
    "y_units": y_units,
               ^^^^^^^
UnboundLocalError: cannot access local variable 'y_units' where it is not associated with a value
Aborted (core dumped)
```

Can't reproduce this one any more, must be fixed in the current state of the code.

### LINKED ISSUE #107: BUG: test files, some crash the app
Using _mdaviz_ from _main_ branch, encountered this repeatable error:

```py
========= Selected file: 14BMC_0015.mda in dev_sampler/mda

doPlot called: args=('replace', {'Y': [], 'X': 1})
Traceback (most recent call last):
  File "projects/BCDA-APS/mdaviz/mdaviz/mda_folder.py", line 309, in doFileSelected
    self.handlePlotBasedOnMode()
  File "projects/BCDA-APS/mdaviz/mdaviz/mda_folder.py", line 332, in handlePlotBasedOnMode
    self.doPlot("replace", self.selectionField())
  File "projects/BCDA-APS/mdaviz/mdaviz/mda_folder.py", line 357, in doPlot
    datasets, plot_options = to_datasets(
                             ^^^^^^^^^^^^
  File "/home/prjemian/Documents/projects/BCDA-APS/mdaviz/mdaviz/select_fields_table_view.py", line 241, in to_datasets
    "y_units": y_units,
               ^^^^^^^
UnboundLocalError: cannot access local variable 'y_units' where it is not associated with a value
```

Here: https://github.com/BCDA-APS/mdaviz/blob/d6e04896d3f8f73fa0a8d7a171861ed058b6522b/mdaviz/select_fields_table_view.py#L241

Perhaps the `for ...` loop at line 221, which includes assigned `y_units`, is not run in this case? https://github.com/BCDA-APS/mdaviz/blob/d6e04896d3f8f73fa0a8d7a171861ed058b6522b/mdaviz/select_fields_table_view.py#L221-L225

File in attached zip: [14BMC_0015.zip](https://github.com/BCDA-APS/mdaviz/files/14748347/14BMC_0015.zip)

### LINKED ISSUE #124: Performance issue with /net/sXXdata
Opening & loading folders on /net/s29data can be extremely slow. 
@prjemian could you help me with that? I am not sure how to troubleshoot and improve this kind of things...

### PR DESCRIPTION
**Pull request by rodolakis**
- close #86
- close #124

- partially close #107

### CODE REVIEW

File: mdaviz/app.py [hunk 1]
```diff
@@ -74,16 +72,12 @@ def main():  # for future command-line options
 
     # Ensure the path is absolute (starts with a "/")
     if not directory.startswith("/"):
```
Comment (prjemian) [inline mdaviz/app.py:74]: This `.startswith("/")` makes the program depend on specific filesystems.  Won't work on Windows.

File: mdaviz/mainwindow.py [hunk 2]
```diff
@@ -17,71 +16,62 @@
 from .opendialog import DIR_SETTINGS_KEY
 
 UI_FILE = utils.getUiFileName(__file__)
+MAX_FILES = 100
```
Comment (prjemian) [inline mdaviz/mainwindow.py:19]: -#57 

For more than NNN files, could show a dialog with a progress bar and a Cancel button.  As a UI, this leaves users the option to bail out if it takes too long.  (Add as enhancement issue).

Comment (prjemian) [reply mdaviz/mainwindow.py:19]: The `tqdm` package is great for progress bars.

File: mdaviz/app.py (hunk 1 above)
Comment (prjemian) [reply mdaviz/app.py:74]: Maybe compare `str(directory.absolute())` with `str(directory)`?

**Review by prjemian — APPROVED** [review]
Great!

File: mdaviz/app.py (hunk 1 above)
Comment (rodolakis) [reply mdaviz/app.py:74]: created #141

File: mdaviz/app.py [hunk 3]
```diff
@@ -23,9 +23,7 @@ def gui(directory):
 
     app = QtWidgets.QApplication(sys.argv)
     main_window = MainWindow(directory=directory)
-    main_window.setStatus(
-        f"Application started, loading {pathlib.Path(directory).absolute()} ..."
-    )
+    main_window.setStatus(f"Application started, loading {pathlib.Path(directory).absolute()} ...")
```
[the remark this answers is not in the document]
Comment (rodolakis) [reply mdaviz/app.py:26]: created https://github.com/BCDA-APS/mdaviz/issues/141

File: mdaviz/mainwindow.py (hunk 2 above)
Comment (rodolakis) [reply mdaviz/mainwindow.py:19]: created #142

**Pull request merged**